Document function application operator (#11305)
NixOS/nix: doc/manual/src/language/operators.md

[Attribute selection]: #attribute-selection

## Function application

> **Syntax**
>
> *func* *expr*

> Apply the callable value *func* to the argument *expr*. Note the absence of a visible operator symbol.

Reviewer: ```suggestion
Apply the callable value *func* to the argument *expr*. Note the absence of an operator symbol.
```

Reviewer 2: ```suggestion
Apply the callable value *func* to the argument *expr*. Note the absence of any visible operator symbol.
```

A callable value is either:
- a [user-defined function][function]
- a [built-in][builtins] function
- an attribute set with a [`__functor` attribute](./syntax.md#attr-__functor)

> **Warning**
>

> > This "operator" also separates [list] items, which means that calls in list items must be enclosed by parentheses.

Reviewer: ```suggestion
> [List][list] items are also separated by whitespace, which means that function calls in list items must be enclosed by parentheses.
```
